server/server.ts

```diff
@@ -0,0 +1,191 @@
+import bcrypt from 'bcryptjs'
+import { Request, Response, NextFunction } from 'express'
+import jsonServer from 'json-server'
+import jwt from 'jsonwebtoken'
+import { dirname } from 'path'
+import { fileURLToPath } from 'url'
+
+import { DbUser, convertDbUserToUser } from './dbModels'
+import { User, Post, Gender } from '../src/models'
+
+// Type for the database schema
+interface Database {
+  users: DbUser[]
+  posts: Post[]
+}
+
+// Get the directory name of the current module file (for db.json path)
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = dirname(__filename)
+
+// Create a new jsonServer instance
+const server = jsonServer.create()
+
+// Middleware to parse JSON bodies
+server.use(jsonServer.bodyParser)
+
+// Assuming `router.db` is a Lowdb instance with a JSON backend
+const router = jsonServer.router<Database>(`${__dirname}/db.json`)
+// Use default middlewares (for CORS, logging, etc.)
+const middlewares = jsonServer.defaults()
+
+interface RegisterRequestBody {
```

Comment:
Please move the interfaces along with the other interface.

Each file should have the following structure:

Interfaces
const declarations if present
functions.